File: daily/January/739.java
import java.util.Stack;

// Read solution
// Look up monotonic stack more
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> monoStack = new Stack<>();
        int[] res = new int[temperatures.length];

        for (int i = temperatures.length - 1; i >= 0; i--) {
            while (!monoStack.empty() && temperatures[i] >= temperatures[monoStack.peek()]) {
                monoStack.pop();
            }
            if (!monoStack.empty()) {
                res[i] = monoStack.peek() - i;
            }
            monoStack.push(i);
        }
        return res;
    }
}

File: Leetcode75/739.cs
namespace DailyTemperatures;
public class Solution
{
    public int[] DailyTemperatures(int[] temperatures)
    {
        Stack<(int temp, int idx)> monoStack = new();
        int[] res = new int[temperatures.Length];
        for (int i = temperatures.Length - 1; i >= 0; i--)
        {
            int wait = 0;
            while (monoStack.Any() && monoStack.Peek().temp <= temperatures[i])
            {
                monoStack.Pop();
                wait++;
            }
            res[i] = monoStack.Any() ? monoStack.Peek().idx - i: 0;
            monoStack.Push((temperatures[i], i));
        }
        return res;
    }
}
